openlibrary/plugins/upstream/account.py

```diff
@@ -769,6 +776,51 @@ def GET(self, username, key='loans'):
         raise web.seeother(user.key)
 
 
+class public_my_books_json(delegate.page):
+    encoding = "json"
+    path = "/people/([^/]+)/books/([a-zA-Z_-]+)"
```

Comment:
For my public account `mekBot` http://testing.openlibrary.org/people/mekBot/books/want-to-read.json

I get:
```
{
error: "Shelf want-to-read not found or not accessible"
}
```
Yet you may see my public reading log here:
http://testing.openlibrary.org/people/mekBot/books/want-to-read

Reply:
I was puzzled, because I just copied the access control code from the web version of the interface. Then I spotted the unmatched paren I had inserted after testing to abide by the 88 character limit that the preexisting code didn't honor. You should find that it's working now.